.github/workflows/build.yaml

```diff
@@ -0,0 +1,38 @@
+name: Main Build
+
+on:
+    workflow_dispatch:
+    pull_request:
+        branches: [main]
+env:
+  GPG_PASSPHRASE: ${{secrets.GPG_PASSPHRASE}}
+  GPG_PRIVATE_KEY: ${{secrets.GPG_PRIVATE_KEY}}
+
+jobs:
+    build-push-wrapping-artefacts:
+        name: Building and Deploying Maven artefcats
+        runs-on: ubuntu-latest
+
+        steps:
+          - name: Checkin Code
+            uses: actions/checkout@v4
+          
+          - name: Set up JDK 11
+            uses: actions/setup-java@v4
+            with:
+              java-version: '11'
+              distribution: 'temurin'
+              gpg-private-key: ${{secrets.GPG_PRIVATE_KEY}}
+              gpg-passphrase: GPG_PASSPHRASE
+              cache: maven
+
+          - name: Building artefacts
+            run: |
+             cat ~/.m2/settings.xml
+             mvn --batch-mode -Dgpg.passphrase=${{secrets.GPG_PASSPHRASE}} clean deploy -Dgalasa.source.repo=https://repo.maven.apache.org/maven2/ \
+             -Dgalasa.central.repo=https://repo.maven.apache.org/maven2/   \
+             -Dgalasa.release.repo=https://maven.pkg.github.com/jaydee029/wrapping        \
+             -Dgalasa.jacocoEnabled=true -Dgalasa.isRelease=false
+            env:
+              GPG_PASSPHRASE: ${{secrets.GPG_PASSPHRASE}}

```

Comment:
- I believe the `--errors --fail-at-end` flags are needed here as well as `--batch-mode`
- I'm not sure the `clean` command is needed as well as the `deploy`
- `-Dgalasa.isRelease` should be `true`
- This should be pointing at a settings.xml file with the `--settings` flag - the settings.xml file in our current Tekton builds is generated in the [mavengpg Task](https://github.com/galasa-dev/automation/blob/main/pipelines/tasks/maven-gpg.yaml) which uses a Kubernetes [ExternalSecret mavengpg](https://github.com/galasa-dev/automation/blob/main/infrastructure/cicsk8s/galasa-build/external-secrets/secret-mavengpg.yaml). We'll have to discuss how to replicate this in GH actions.